spec/models/validations/property_validations_spec.rb

```diff
@@ -4,194 +4,222 @@
       it "does not add an error" do
-        record.beds = 2
-        record.unittype_gn = nil
-        property_validator.validate_shared_housing_rooms(record)
-        expect(record.errors).to be_empty
+        log.beds = 2
+        log.unittype_gn = nil
+        property_validator.validate_shared_housing_rooms(log)
+        expect(log.errors).to be_empty
       end
     end
 
     context "when a bedsit has more than 1 bedroom" do
-      let(:expected_error) { I18n.t("validations.property.unittype_gn.one_bedroom_bedsit") }
+      before do
+        log.beds = 2
+        log.unittype_gn = 2
+      end
 
-      it "adds an error" do
-        record.beds = 2
-        record.unittype_gn = 2
-        property_validator.validate_shared_housing_rooms(record)
-        expect(record.errors["unittype_gn"]).to include(match(expected_error))
-        expect(record.errors["beds"]).to include(I18n.t("validations.property.unittype_gn.one_bedroom_bedsit"))
+      context "and the log is for 24/25 or later" do
+        it "does not add an error" do
+          property_validator.validate_shared_housing_rooms(log)
+
+          expect(log.errors).to be_empty
+        end
+      end
+
+      context "and the log is from before 24/25" do
+        it "adds an error" do
+          allow(log.form).to receive(:start_year_after_2024?).and_return false
+
+          property_validator.validate_shared_housing_rooms(log)
+
+          expect(log.errors["unittype_gn"]).to include(I18n.t("validations.property.unittype_gn.one_bedroom_bedsit"))
+          expect(log.errors["beds"]).to include(I18n.t("validations.property.unittype_gn.one_bedroom_bedsit"))
+        end
       end
     end
 
     context "when a bedsit has less than 1 bedroom" do
-      let(:expected_error) { I18n.t("validations.property.unittype_gn.one_bedroom_bedsit") }
+      before do
+        log.beds = 0
+        log.unittype_gn = 2
+      end
 
-      it "adds an error" do
-        record.beds = 0
-        record.unittype_gn = 2
-        property_validator.validate_shared_housing_rooms(record)
-        expect(record.errors["unittype_gn"]).to include(match(expected_error))
-        expect(record.errors["beds"]).to include(I18n.t("validations.property.unittype_gn.one_bedroom_bedsit"))
+      context "and the log is for 24/25 or later" do
+        it "does not add an error" do
+          property_validator.validate_shared_housing_rooms(log)
+
+          expect(log.errors).to be_empty
+        end
+      end
+
+      context "and the log is from before 24/25" do
+        it "adds an error" do
+          allow(log.form).to receive(:start_year_after_2024?).and_return false
+
+          property_validator.validate_shared_housing_rooms(log)
+
+          expect(log.errors["unittype_gn"]).to include(I18n.t("validations.property.unittype_gn.one_bedroom_bedsit"))
+          expect(log.errors["beds"]).to include(I18n.t("validations.property.unittype_gn.one_bedroom_bedsit"))
+        end
```

Comment:
whether or not errors are added in this case was changed between years in 3135 but not tested.